subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -202,13 +248,51 @@ int img_mgmt_get_next_boot_slot(int image, enum img_mgmt_next_boot_type *type)
 	int rcs = img_mgmt_read_info(other_slot, &over, NULL, NULL);
 	int rca = img_mgmt_read_info(active_slot, &aver, NULL, NULL);
 
+#if defined(CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT)
+	active_slot_state = read_directxip_state(active_slot);
+	other_slot_state = read_directxip_state(other_slot);
+	if (rca != 0 || rcs != 0 || other_slot_state < 0 || active_slot_state < 0) {
+		LOG_ERR("Active slot state read = %d, other slot state read %d\n",
+			active_slot_state, other_slot_state);
+		LOG_ERR("Slot version read rc = %d and %d\n", rca, rcs);
+		/* We do not really know what will happen, as we can not
+		 * read states from bootloader.
+		 */
+		return active_slot;
+	}
+
+	if (active_slot_state == DIRECT_XIP_BOOT_REVERT) {
+		*type = NEXT_BOOT_TYPE_REVERT;
```

Comment:
This should align with https://github.com/zephyrproject-rtos/zephyr/blob/7fc74285c490ecf852edced1a9ebc37b36084ea2/subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c#L194-L196 whereby it uses a local variable and only updates the user value if they provided a pointer and it's not null

Reply:
Fixed